utilities/src/serialize/impls.rs

```diff
@@ -148,7 +148,46 @@ impl_canonical_serialization_uint!(u8);
 impl_canonical_serialization_uint!(u16);
 impl_canonical_serialization_uint!(u32);
 impl_canonical_serialization_uint!(u64);
-impl_canonical_serialization_uint!(usize);
+
+impl CanonicalSerialize for usize {
+    #[inline]
+    fn serialize_with_mode<W: Write>(&self, mut writer: W, _compress: Compress) -> Result<(), SerializationError> {
+        let u64_value = *self as u64;
+        Ok(writer.write_all(&u64_value.to_le_bytes())?)
```

Comment:
Why not 1-line this function?

Something like:
Ok(writer.write_all(&(*self as u64).to_le_bytes())?)